api/controllers/game/spectate.js

```diff
@@ -1,17 +1,44 @@
 module.exports = async function (req, res) {
   try {
     const { gameId } = req.body;
-    const game = await gameService.populateGame({ gameId });
+    const promiseGame = await gameService.populateGame({ gameId });
+    const promisePlayer = await userService.findUser({ userId: req.session.usr });
 
-    if (game.status || game.players.length < 2) {
-      return res.badRequest({message: 'You can only spectate an ongoing game with two players'});
-    }
-
-    // Subscribe socket to game
-    Game.subscribe(req, [ gameId ]);
-    req.session.spectating = gameId;
-
-    return res.ok(game);
+    Promise.all([promiseGame, promisePlayer])
+      .then(function checkAndSubscribe(values) {
```

Comment:
Earlier in this function we're using `async/await` rather than `Promise.then()`; these are alternatives to each other and it's best to use one or the other without mixing them. Generally `async/await` is the newer syntax and is our preference for this project when writing new code.

`async/await` syntax can still be used with Promise.all() and would be cleaner here:

```
const [ game, spectator ] = await Promise.all([ promiseGame, promiseSpectator ]);
// Now you can use `game` and `spectator`
```